src/compiler.rs

```diff
@@ -75,14 +86,65 @@ impl Compiler {
               .join(Self::expand_tilde(&relative.cooked)?)
               .lexiclean();
 
-            if import.is_file() {
+            let has_glob = import.as_os_str().as_encoded_bytes().contains(&b'*');
+
+            if has_glob {
+              if !(unstable || ast_unstable) {
+                return Err(Error::UnstableFeature {
+                  unstable_feature: UnstableFeature::GlobImport,
+                });
+              }
+
+              let import_base_str =
+                import
+                  .as_os_str()
+                  .to_str()
+                  .ok_or_else(|| Error::ImportGlob {
+                    error: "non-UTF-8 glob import".to_string(),
+                    path: *path,
+                  })?;
+              let glob_options = glob::MatchOptions {
+                case_sensitive: true,
+                require_literal_separator: false,
+                require_literal_leading_dot: false,
+              };
+              let import_paths =
+                glob::glob_with(import_base_str, glob_options).map_err(|error| {
+                  Error::ImportGlob {
```

Comment:
```suggestion
                  Error::ImportGlobPattern {
```

I believe this top level glob error only happens if the pattern is bad.